lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php

```diff
@@ -48,17 +50,17 @@ class MysqliConnection implements Connection, PingableConnection, ServerInfoAwar
      */
     public function __construct(array $params, $username, $password, array $driverOptions = [])
     {
-        $port = $params['port'] ?? ini_get('mysqli.default_port');
+        $port = (int) ($params['port'] ?? ini_get('mysqli.default_port'));
```

Comment:
Let's only cast the value from `ini_get()`. If the explicitly specified port parameter is non-int, let it fail. `??` could be probably replaced with `?:`.

Reply:
How about when the value comes i.e. from XML?

Reply:
That's up to the caller to fix, not the callee. The definition of `$params['port']` requires an `int`, although that isn't expressed in the docblock (`mixed[]`).

We could improve that with further declarations...

Reply:
Removed and changed on caller part.